source/going_further.rst

```diff
@@ -0,0 +1,472 @@
+#############
+Going Further
+#############
+
+This section shows some helpful tools that might aid you along your programming path. We briefly describe here how
+to structure larger projects and work with multiple source files. We discuss python coding style guide (also known as PEP8).
+and how pylint can help you to write and maintain readable and stylish code. At the end there is a list of resources you might find
+helpful in your day-to-day python programming.
+
+
+Managing larger projects
+========================
+
+Divide and conquer
+--------------------
+
+Up to this point, you might have put all of your source coed in one file. For small libraries and project this is fine, but when complexity
+rises it is better to divide your code into smaller files, that is to create libraries or modules. It is not necessary, for example, to put
+every class in a separate file, but having one file for classes that are logically connected is better for readability.
+Going through a file that has thousands lines of code can be difficult.
+
+As an example, let us consider a program for handling regular polygons. Imagine that every figure has its own, simplified class. You might want to
+consider separating main functionality e. g. calculating perimeter, from definitions of those shapes [#fn1]_. That is, to put calculations of fields in one
+file and definitions in another:
+
+.. code-block:: python
+    
+    # File polygons.py
+    class Circle:
+        def __init__(self, radius):
+            self.no_sides = 0
+            self.radius = radius
+
+    class Triangle:
+        def __init__(self, side):
+            self.no_sides = 3
+            self.side = side
+
+    class Square:
+        def __init__(self):
+            self.no_sides = 4
+            self.side = side
+
+.. code-block:: python
+
+    # File shapefunctions.py
+    PI = 3.141592 # or use math.PI
+    def calc_perimeter(obj):
+        if obj.no_sides > 0:
+            perimeter = obj.no_sides * obj.side
+        else:  
+            perimeter = 2*PI*obj.radius
+        return perimeter
+
+Importing functionalities
+-------------------------
+
+To use definitions and perimeter function you must now import those functionalities. Code below asks user to specify which shape to
+use and for the side length/radius, and calculates perimeter:
+
+.. code-block:: python
+
+    # Main file or command line
+    import polygons
+    import shapefunctions
+
+    print("Choose a shape:")
+    print("1. Circle")
+    print("2. Triangle")
+    print("3. Square")
+    choice = int(input())
+    print("Choose the side length")
```

Comment:
Can just be 
```
choice = int(input("Choose the side length"))
```

Reply:
Also, I would change it to "Choose the side length or radius"